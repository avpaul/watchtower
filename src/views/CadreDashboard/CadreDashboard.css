/* /*
 Container
 */

.cadre-submenu {
  margin: auto;
}

.cadre__page {
  border-left: 1px solid rgba(216, 216, 216, 0.45);
  padding-left: 25px;
}

/*
  Card css
 */

.cadre-submenu-cards {
  cursor: pointer;
  height: 108px;
  width: 240px;
  margin-bottom: 22px;
  border-radius: 6px;
  background-color: #ffffff;
  box-shadow: 2px 4px 10px 0 #d8d8d8;
}

.project-badge {
  height: 31px;
  margin-left: 0.3em;
  text-align: center;
  width: 31px;
  background-color: white;
  border-radius: 50%;
}

.submenu-title {
  height: 23px;
  width: 80px;
  margin-left: 0.3em;
  color: #333e44;
  font-family: 'DIN Pro';
  font-size: 18px;
  line-height: 23px;
}

.submenu-subtitle {
  height: 21px;
  width: 200px;
  color: rgba(51, 62, 68, 0.51);
  font-family: 'DIN Pro';
  font-size: 16px;
  padding-left: 20px;
  line-height: 21px;
}

.cards-header {
  padding-top: 21px;
  padding-left: 18px;
}
.cadre-page-content {
  padding-top: 10%;
  padding-left: 0;
  padding-right: 9%;
  margin-left: 0;
}
.iconImage {
  height: 17px;
  width: 17px;
}

/*
Media queries - Responsiveness
 */

@media (min-width: 300px) and (max-width: 540px) {
  .nav-stacked {
    display: table;
    margin: 0 auto;
  }
}

@media (min-width: 1365px) {
  .nav-stacked {
    display: table;
    margin: 0 auto;
  }

  .col-sm-3 {
    max-width: 20%;
  }

  .col-sm-9 {
    max-width: 80%;
  }
}

@media (min-width: 1100px) and (max-width: 1365px) {
  .nav-stacked {
    display: table;
    margin: 0 auto;
  }

  .col-sm-3 {
    max-width: 25%;
  }

  .col-sm-9 {
    max-width: 75%;
  }
}

@media (min-width: 540px) and (max-width: 575px) {
  .cadre-submenu-cards {
    margin-right: 15px;
  }
}

@media (max-width: 1023px) and (min-width: 576px) {
  [class*='col-'] {
    max-width: 100%;
  }
}
