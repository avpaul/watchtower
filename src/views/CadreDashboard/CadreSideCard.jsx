import React from 'react';
import sidebarIcon from '../../static/sidebar.png';

const CadreSideCard = () => (
  <div className="cadre-side-card">
    <img
      src={sidebarIcon}
      alt="sidebar icon"
      className="cadre-side-card-icon"
    />
  </div>
);

export default CadreSideCard;
