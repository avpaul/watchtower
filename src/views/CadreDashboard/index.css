/* Cadre Dashboard */
.cadre-content {
  background: #dae7f8;
}

.cadre-side-card {
  height: 7.4rem;
  width: 7.4rem;
  border-radius: 1.25rem;
  background-color: #ffffff;
  box-shadow: 0 0.1rem 0.7rem 0 rgba(0, 0, 0, 0.03);
  margin-left: 2rem;
  margin-top: 2rem;
}

.cadre-side-card:hover {
  cursor: pointer;
}

.cadre-side-card .cadre-side-card-icon {
  height: 63px;
  width: 63px;
  border-radius: 31.5px;
  background-color: #edf4fd;
  margin: 1.6rem 1.9rem 1.9rem 1.6rem;
}
