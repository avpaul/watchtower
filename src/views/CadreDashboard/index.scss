.cadre-content {
  display: flex;
}

.dashboard-wrapper {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-left: 4%;
  margin-right: 4%;
}

.dashboard-greeting {
  margin-top: 40px;
  margin-bottom: 1rem;
}

.fellow-dashboard-main {
  display: flex;
  width: 100%;
  justify-content: space-between;
}

.cadre-side-card-dashboard {
  height: 7.4rem;
  width: 7.4rem;
  border-radius: 1.25rem;
  background-color: #ffffff;
  box-shadow: 0 0.1rem 0.7rem 0 rgba(0, 0, 0, 0.03);
  margin-left: 2rem;
  margin-top: 2rem;
}

.cadre-side-card:hover {
  cursor: pointer;
}

.cadre-side-card .cadre-side-card-icon {
  height: 63px;
  width: 63px;
  border-radius: 31.5px;
  background-color: #edf4fd;
  margin: 1.6rem 1.9rem 1.9rem 1.6rem;
}
