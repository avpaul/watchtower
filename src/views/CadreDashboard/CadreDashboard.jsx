import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Switch, Route } from 'react-router-dom';
import CadreSubmenu from '../../components/CadreSubmenu';
import ProjectsDashboard from './ProjectsDashboard';
import RolesDashboard from './RolesDashboard';
import ReportsDashboard from './ReportsDashboard';
import VacanciesDashboard from './VacanciesDashboard';
import CertificationsDashboard from './CertificatesDashboard';

import './CadreDashboard.scss';
import NotFoundPage from '../NotFoundPage';

class CadreDashboard extends Component {
  /**
   * Creates the CadreDashboard Component and initializes state
   * @constructor
   * @param {*} props - Super props inherited by Component
   */
  constructor(props) {
    super(props);
    this.state = {
      routes: [
        `${props.match.url}/projects`,
        `${props.match.url}/vacancies`,
        `${props.match.url}/roles`,
        `${props.match.url}/certifications`,
        `${props.match.url}/reports`
      ]
    };
  }

  /**
   * Listens to events specifically click events
   *  @param {*} event
   */
  handleCardclick = event => {
    const { history } = this.props;
    const { routes } = this.state;
    const id = event.target.getAttribute('data-key');
    if (routes[id]) history.replace(routes[id]);
  };

  renderSubMenu = () => {
    const { routes } = this.state;
    const { location } = this.props;
    const routeIndex = routes.findIndex(
      route => location.pathname.search(route) === 0
    );
    return (
      <div className="col-sm-3 sidebar-submenu">
        <CadreSubmenu
          handleCardclick={this.handleCardclick}
          activeItem={`${routeIndex}`}
          submenuType="opsSubmenus"
        />
      </div>
    );
  };

  renderRoute = (RouteComponent, url) => (
    <Route
      path={url}
      component={newProps => (
        <RouteComponent {...{ ...this.props, match: newProps.match }} />
      )}
    />
  );

  render() {
    const { routes } = this.state;
    return (
      <div
        id="cadre-dashboard-container"
        className="container-fluid page-content"
      >
        <div className="row">
          {this.renderSubMenu()}
          <div className="col-sm-9 main-content">
            <div>
              <Switch>
                {this.renderRoute(ProjectsDashboard, routes[0])}
                {this.renderRoute(VacanciesDashboard, routes[1])}
                {this.renderRoute(RolesDashboard, routes[2])}
                {this.renderRoute(CertificationsDashboard, routes[3])}
                {this.renderRoute(ReportsDashboard, routes[4])}
                <Route component={NotFoundPage} />
              </Switch>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

CadreDashboard.propTypes = {
  history: PropTypes.shape().isRequired,
  match: PropTypes.shape().isRequired,
  location: PropTypes.shape().isRequired
};

export default CadreDashboard;
