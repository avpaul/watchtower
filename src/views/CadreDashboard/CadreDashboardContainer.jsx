import CadreDashboard from './CadreDashboard';

export default CadreDashboard;
