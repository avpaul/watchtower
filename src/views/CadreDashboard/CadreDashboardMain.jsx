import React from 'react';
import CadreSideCard from './CadreSideCard';
import './index.css';

const CadreDashboardMain = () => (
  <div className="cardre-content">
    <CadreSideCard />
  </div>
);

export default CadreDashboardMain;
