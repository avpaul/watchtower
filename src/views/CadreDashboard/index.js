import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from '../../components/Header';
import CadreDashboardMain from './CadreDashboardMain';

const CadreDashboard = props => (
  <Route path="/dashboard">
    <React.Fragment>
      <Header {...props} />
      <Switch>
        <Route
          exact
          path="/dashboard"
          component={() => <CadreDashboardMain {...props} />}
        />
        <Header {...props} />
      </Switch>
    </React.Fragment>
  </Route>
);

export default CadreDashboard;
