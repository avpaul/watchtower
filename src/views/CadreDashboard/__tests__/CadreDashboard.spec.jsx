import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import waitForExpect from 'wait-for-expect';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Thunk from 'redux-thunk';

import CadreDashboard from '../CadreDashboard';
import initialState from '../../../redux/reducers/initialState';

describe('Tests the CadreDashboard component', () => {
  const defaultProps = {
    match: {
      url: '/cadre'
    },
    history: {
      replace: jest.fn()
    },
    location: {
      pathname: '/cadre/projects'
    }
  };

  it('should render without crushing', () => {
    const wrapper = shallow(<CadreDashboard {...defaultProps} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('should execute handleCardclick method', () => {
    const mockStore = configureStore([Thunk]);
    const store = mockStore(initialState);

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter keyLength={0} initialEntries={['/cadre/reports']}>
          <CadreDashboard {...defaultProps} />
        </MemoryRouter>
      </Provider>
    );

    const event = {
      target: {
        getAttribute: () => '0'
      }
    };

    wrapper
      .find('.cadre-submenu-cards')
      .first()
      .simulate('click', event);

    waitForExpect(() => {
      expect(wrapper.find('ProjectsDashboard').exists()).toBeTruthy();
    });
  });
});
