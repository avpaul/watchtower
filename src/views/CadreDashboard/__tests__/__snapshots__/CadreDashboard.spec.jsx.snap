// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the CadreDashboard component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CadreDashboard
    history={
        Object {
            "replace": [Function],
          }
    }
    location={
        Object {
            "pathname": "/cadre/projects",
          }
    }
    match={
        Object {
            "url": "/cadre",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="container-fluid page-content"
>
        <Connect(AddManagerModal) />
        <Connect(AddTechnologyModal) />
        <div
                className="row"
        >
                <div
                        className="col-sm-3 sidebar-submenu"
                >
                        <CadreSubmenu
                                activeItem="0"
                                handleCardclick={[Function]}
                        />
                </div>
                <div
                        className="col-sm-12 col-lg-9"
                >
                        <Switch>
                                <Route
                                        component={[Function]}
                                        path="/cadre/projects"
                                />
                        </Switch>
                </div>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(AddManagerModal) />,
          <Connect(AddTechnologyModal) />,
          <div
            className="row"
>
            <div
                        className="col-sm-3 sidebar-submenu"
            >
                        <CadreSubmenu
                                    activeItem="0"
                                    handleCardclick={[Function]}
                        />
            </div>
            <div
                        className="col-sm-12 col-lg-9"
            >
                        <Switch>
                                    <Route
                                                component={[Function]}
                                                path="/cadre/projects"
                                    />
                        </Switch>
            </div>
</div>,
        ],
        "className": "container-fluid page-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-sm-3 sidebar-submenu"
>
                <CadreSubmenu
                                activeItem="0"
                                handleCardclick={[Function]}
                />
</div>,
              <div
                className="col-sm-12 col-lg-9"
>
                <Switch>
                                <Route
                                                component={[Function]}
                                                path="/cadre/projects"
                                />
                </Switch>
</div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CadreSubmenu
                  activeItem="0"
                  handleCardclick={[Function]}
/>,
                "className": "col-sm-3 sidebar-submenu",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeItem": "0",
                  "handleCardclick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Switch>
                  <Route
                                    component={[Function]}
                                    path="/cadre/projects"
                  />
</Switch>,
                "className": "col-sm-12 col-lg-9",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Route
                    component={[Function]}
                    path="/cadre/projects"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/cadre/projects",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="container-fluid page-content"
>
          <Connect(AddManagerModal) />
          <Connect(AddTechnologyModal) />
          <div
                    className="row"
          >
                    <div
                              className="col-sm-3 sidebar-submenu"
                    >
                              <CadreSubmenu
                                        activeItem="0"
                                        handleCardclick={[Function]}
                              />
                    </div>
                    <div
                              className="col-sm-12 col-lg-9"
                    >
                              <Switch>
                                        <Route
                                                  component={[Function]}
                                                  path="/cadre/projects"
                                        />
                              </Switch>
                    </div>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(AddManagerModal) />,
            <Connect(AddTechnologyModal) />,
            <div
              className="row"
>
              <div
                            className="col-sm-3 sidebar-submenu"
              >
                            <CadreSubmenu
                                          activeItem="0"
                                          handleCardclick={[Function]}
                            />
              </div>
              <div
                            className="col-sm-12 col-lg-9"
              >
                            <Switch>
                                          <Route
                                                        component={[Function]}
                                                        path="/cadre/projects"
                                          />
                            </Switch>
              </div>
</div>,
          ],
          "className": "container-fluid page-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-sm-3 sidebar-submenu"
>
                  <CadreSubmenu
                                    activeItem="0"
                                    handleCardclick={[Function]}
                  />
</div>,
                <div
                  className="col-sm-12 col-lg-9"
>
                  <Switch>
                                    <Route
                                                      component={[Function]}
                                                      path="/cadre/projects"
                                    />
                  </Switch>
</div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <CadreSubmenu
                    activeItem="0"
                    handleCardclick={[Function]}
/>,
                  "className": "col-sm-3 sidebar-submenu",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeItem": "0",
                    "handleCardclick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Switch>
                    <Route
                                        component={[Function]}
                                        path="/cadre/projects"
                    />
</Switch>,
                  "className": "col-sm-12 col-lg-9",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Route
                      component={[Function]}
                      path="/cadre/projects"
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/cadre/projects",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
